platform/ui/src/contextProviders/DialogProvider.js

```diff
@@ -114,62 +125,84 @@ const DialogProvider = ({ children, service }) => {
     });
   };
 
+  const validCallback = callback => callback && typeof callback === 'function';
+
   return (
-    <DialogContext.Provider value={{ create, dismiss, dismissAll, dialogs }}>
+    <DialogContext.Provider value={{ create, dismiss, dismissAll, noDialogs }}>
       <div className="DraggableArea">
         {dialogs.map(dialog => {
           const {
-            id,
-            content: Dialog,
-            position /* Position of the dialog. {{x: 0, y: 0}} */,
-            defaultPosition,
+            id = null,
+            content: DialogContent = null,
+            contentProps = null,
+            position = null,
+            defaultPosition = null,
             isDraggable = true,
-            onStart = () => {},
-            onStop = () => {},
-            onDrag = () => {},
+            showOverlay = false,
+            onStart = null,
+            onStop = null,
+            onDrag = null,
           } = dialog;
 
           return (
-            <Draggable
+            <div
               key={id}
-              disabled={!isDraggable}
-              position={position}
-              defaultPosition={lastDialogPosition || defaultPosition}
-              bounds="parent"
-              onStart={event => {
-                const e = event || window.event;
-                const target = e.target || e.srcElement;
-                const BLACKLIST = ['SVG', 'BUTTON', 'PATH', 'INPUT'];
-                if (BLACKLIST.includes(target.tagName.toUpperCase())) {
-                  return false;
-                }
-
-                onStart(event);
-              }}
-              onStop={event => {
-                onStop(event);
-                setIsDragging(false);
-                return;
-              }}
-              onDrag={event => {
-                setIsDragging(true);
-                _reorder(id);
-                _updateLastDialogPosition(id);
-                onDrag(event);
-              }}
+              className={classNames('Overlay', showOverlay && 'active')}
```

Comment:
You may want "one" of these that is shown if any dialog is attempting to `showOverlay`? Unless we want to support a scenario where you have one dialog, that opens another dialog but hides the previous beneath the overlay to prevent interaction?

Reply:
This overlay is meant to be used if the user has to answer one dialog at a time, like when the user adds a new arrow annotation, we cannot allow the user to just keep creating dialogs before closes the last.

Reply:
If you think we'll need to support more than one overlay, then keep it but document it? If you think we'll only ever need one overlay, bump it up a layer?

Reply:
Updated. Only render overlay if theres at least one dialog with showOverlay prop, thanks.